portfolio/src/main/client/src/components/sections/comments.ts

```diff
@@ -138,23 +143,78 @@ async function submitForm(
   await loadCurrentPage(commentsEl, pages, limit);
 }
 
+/**
+ * @param text The text to truncate.
+ * @returns `text` truncated to 5 lines or 500 chars, whichever is shorter.
+ */
+function truncateText(text: string) {
+  const truncatedByChars = text.slice(0, 500);
+  const truncatedByNewlines = text.split('\n').slice(0, 5).join('\n');
+
+  if (truncatedByChars.length < truncatedByNewlines.length) {
+    return truncatedByChars;
+  } else {
+    return truncatedByNewlines;
+  }
+}
+
 const Comment = (comment: CommentInfo): HTMLElement => {
   // user supplied strings cannot be interpolated directly to avoid XSS
-  const content: HTMLElement = htmlElement`<div class="content"></div>`;
-  content.innerText = comment.content;
+  const contentEl: HTMLElement = htmlElement`<div class="content"></div>`;
+  contentEl.innerText = comment.content;
+
+  let truncatedContentEl: HTMLElement | null = null;
+  let truncatedContentExpander: HTMLAnchorElement | null = null;
+  const truncatedContent = truncateText(comment.content);
+
+  if (truncatedContent.length < comment.content.length) {
+    truncatedContentEl = document.createElement('div');
+    truncatedContentEl.classList.add('content', 'truncated');
+    truncatedContentEl.innerText = truncatedContent;
+
+    truncatedContentExpander = document.createElement('a');
+    truncatedContentExpander.href = 'javascript:void 0';
+    truncatedContentExpander.classList.add('expander');
+    truncatedContentExpander.innerText = 'see more';
+
+    let isTruncated = true;
+    truncatedContentExpander.addEventListener('click', () => {
+      if (isTruncated) {
+        truncatedContentEl!.replaceWith(contentEl);
+        truncatedContentExpander!.innerText = 'see less';
+      } else {
+        contentEl.replaceWith(truncatedContentEl!);
+        truncatedContentExpander!.innerText = 'see more';
+      }
+
+      isTruncated = !isTruncated;
+    });
+  }
+
 
   const name: HTMLElement = htmlElement`<span class="name"></span>`;
   name.innerText = comment.name;
 
-  return htmlElement`
-  <li class="comment" data-id="${comment.id.toString()}">
-    <span class="info">
-      ${name}
-      <span class="upvotes">+${comment.upvotes.toString()}</span>
-      <span class="downvotes">-${comment.downvotes.toString()}</span>
-    </span>
-    ${content}
-  </li>`;
+  const commentEl: HTMLElement = htmlElement`
+    <li class="comment" data-id="${comment.id.toString()}">
+      <div class="inner">
+        <span class="info">
+          ${name}
+          <span class="upvotes">+${comment.upvotes.toString()}</span>
+          <span class="downvotes">
+            -${comment.shameful ? '∞' : comment.downvotes.toString()}
```

Comment:
nit: use 	&infin; instead of the special character.

Reply:
What? Are you saying I should use an escape code (`\u221e`)?

Reply:
rip github. It parsed the html character. Use `&infin;`

Reply:
Wait, didn't you say earlier that I should use &copy; instead of `&copy;` because the page is encoded in UTF-8 anyway? Why do you now want me to use an HTML entity?

Reply:
I think I recommended that based on the style guidance here: https://google.github.io/styleguide/htmlcssguide.html#Entity_References

@geekster777 what do you think?